### PR TITLE
274: Fixed histogram rendering

### LINKED ISSUE #274: Pixel-perfect rendering
The library has a lot of small rendering issues, generally on HiDPI devices. We have to migrate to [fancy canvas library](https://github.com/tradingview/fancy-canvas) in order of fixing it.

### PR DESCRIPTION
**Pull request by eugene-korobko**
**Type of PR:** enhancement

**PR checklist:**

- [x] Addresses an existing issue: fixes #274
- [x] Includes tests
- [ ] Documentation update

**Overview of change:**
Make histogram drawing items with fillRect, also fixed pixel alignment

### CODE REVIEW

File: src/renderers/histogram-renderer.ts [hunk 1]
```diff
@@ -34,20 +33,23 @@ export class PaneRendererHistogram implements IPaneRenderer {
...
 			const y = item.y as number;
 			const left = item.left as number;
 			const right = item.right as number;
-			ctx.rect(left, y, right - left, histogramBase - y);
+			ctx.fillRect(left, y, right - left, histogramBase - y);
 		}
-		ctx.fill();
+
+		// TODO: remove this after removing global translate
+		ctx.translate(0.5, +0.5);
```
Comment (timocov) [inline src/renderers/histogram-renderer.ts]:
```suggestion
		ctx.translate(0.5, 0.5);
```

File: src/renderers/histogram-renderer.ts [hunk 2]
```diff
@@ -34,20 +33,23 @@ export class PaneRendererHistogram implements IPaneRenderer {
...
+
```
Comment (timocov) [inline src/renderers/histogram-renderer.ts]:
```suggestion
```

File: src/renderers/histogram-renderer.ts [hunk 3]
```diff
@@ -34,20 +33,22 @@ export class PaneRendererHistogram implements IPaneRenderer {
 
 		const histogramBase = this._data.histogramBase;
 
-		ctx.translate(0.5, 0.5);
+		// TODO: remove this after removing global translate
+		ctx.translate(-0.5, -0.5);
 
 		ctx.fillStyle = this._data.color;
 		ctx.lineWidth = 1;
```
Comment (Nipheris) [inline src/renderers/histogram-renderer.ts]: Do we need this with `fillRect`s?

Comment (timocov) [inline src/renderers/histogram-renderer.ts]:
```suggestion
```

?

Comment (eugene-korobko) [reply src/renderers/histogram-renderer.ts]: Restored drawing with path

**Pull request merged**